src/build/build_step.go

```diff
@@ -39,6 +42,7 @@ func Build(tid int, state *core.BuildState, label core.BuildLabel) {
 		target.SetState(core.Failed)
 		return
 	}
+	metrics.Record(target, time.Since(start))
```

Comment:
The Record function takes that into account; I thought it might be interesting to record which targets fail the most often.